packages/store/tests/execution/dispatch-outside-zone-ngxs-execution-strategy.spec.ts

```diff
@@ -311,16 +311,13 @@ describe('DispatchOutsideZoneNgxsExecutionStrategy', () => {
       class MockModule {}
 
       // Act
-      const warnings: string[] = [];
-
-      console.warn = (...args: string[]) => {
-        warnings.push(args[0]);
-      };
+      const spy = jest.spyOn(console, 'warn').mockImplementation();
 
       await platformBrowserDynamic().bootstrapModule(MockModule, { ngZone: 'noop' });
 
       // Assert
-      expect(warnings).toEqual([CONFIG_MESSAGES[VALIDATION_CODE.ZONE_WARNING]()]);
+      expect(spy).toHaveBeenCalledWith(CONFIG_MESSAGES[VALIDATION_CODE.ZONE_WARNING]());
+      spy.mockRestore();
```

Comment:
This line won't get called if the expectation on the previous line fails.

Reply:
I'm not able to call `mockRestore` before `expect` as Jest releases underneath resources.

Reply:
This is why I suggested wrapping it in a function or as a finally part of a try catch block.

Reply:
Oh, okay :wink:

Reply:
Done